### PR TITLE
plugins: introduce v2.MesonPlugin

### PR DESCRIPTION
**Pull request by sergiusens**
Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

File: snapcraft/plugins/v2/meson.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+            meson_cmd.append(" ".join(self.options.meson_parameters))
+        if not any(c.startswith("--buildtype=") for c in self.options.meson_parameters):
+            meson_cmd.append("--buildtype=${SNAPCRAFT_MESON_BUILDTYPE}")
+        meson_cmd.append(".")
+
+        return [
+            f"python3 -m pip install -U {meson_package}",
+            " ".join(meson_cmd),
+            "ninja",
+            'ninja install DESTDIR="${{SNAPCRAFT_PART_INSTALL}}"',
```
Comment (cjp256) [inline snapcraft/plugins/v2/meson.py:99]: not a format string, the double brackets aren't required

Comment (sergiusens) [reply snapcraft/plugins/v2/meson.py:99]: thanks, I have t fixed locally, but was wrapping up my nodejs->npm plugin :)

Comment (sergiusens) [reply snapcraft/plugins/v2/meson.py:99]: or maybe not :sweat_smile:

Comment (sergiusens) [reply snapcraft/plugins/v2/meson.py]: done

**Pull request merged**